auth_oauth_block_password_login/controllers/main.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 Quartile Limited (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import json
+
+from werkzeug.exceptions import BadRequest
+
+from odoo import SUPERUSER_ID, _, http
+from odoo.http import request
+from odoo.tools.misc import clean_context
+
+from odoo.addons.auth_oauth.controllers.main import (
+    OAuthController as OAuthController,
+    OAuthLogin as Login,
+    fragment_to_query_string,
+)
+from odoo.addons.web.controllers import home
+from odoo.addons.web.controllers.utils import ensure_db
+
+
+class OAuthLogin(Login):
+    @http.route()
+    def web_login(self, *args, **kw):
+        ensure_db()
+        response = super(OAuthLogin, self).web_login(*args, **kw)
+        if response.is_qweb:
+            error = request.params.get("oauth_error")
+            if error == "4":
+                error = _(
+                    "You are not allowed to login with Oauth. Please login with password."
+                )
+                response.qcontext["error"] = error
+        return response
+
+
+class OAuthController(OAuthController):
```

Comment:
I was thinking that domain users would be just redirected to web/login when they access web/reset_password.

But if you have concerns about that, this logic is fine.